File: app/src/main/java/com/rr/rhythms/Services/SettingsService.java
package com.rr.rhythms.Services;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

/**
 * Created by Huruk on 7/10/2016.
 */
public class SettingsService {
    public static void SaveJson(Context context, String key, String json) {
        SharedPreferences prefs = context.getSharedPreferences(key, Context.MODE_PRIVATE);
        SharedPreferences.Editor prefsEditor = prefs.edit();

        prefsEditor
                .putString(key, json)
                .apply();
    }

    public static String ReadJson(Context context, String key) {
        SharedPreferences prefs = context.getSharedPreferences(key, Context.MODE_PRIVATE);
        Gson gson = new Gson();
        return prefs.getString(key, "");
    }

    public static void SaveObject(Context context, String fileName, Object object) {
        try {
            FileOutputStream fos = context.openFileOutput(fileName, Context.MODE_PRIVATE);
            ObjectOutputStream os;
            os = new ObjectOutputStream(fos);
            os.writeObject(object);
            os.close();
            fos.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static Object LoadObject(Context context, String fileName) {
        try {
            FileInputStream fis = context.openFileInput(fileName);
            ObjectInputStream is = new ObjectInputStream(fis);
            Object o = is.readObject();
            is.close();
            fis.close();

            return o;
        } catch (Exception e) {

        }

        return null;
    }
}

File: app/src/main/java/com/rr/rhythms/Services/SettingsService.kt
package com.rr.rhythms.Services

import android.content.Context
import android.content.SharedPreferences

import com.google.gson.Gson

import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException
import java.io.ObjectInputStream
import java.io.ObjectOutputStream

/**
 * Created by Huruk on 7/10/2016.
 */
object SettingsService {
    fun SaveJson(context: Context, key: String, json: String) {
        val prefs = context.getSharedPreferences(key, Context.MODE_PRIVATE)
        val prefsEditor = prefs.edit()

        prefsEditor
                .putString(key, json)
                .apply()
    }

    fun ReadJson(context: Context, key: String): String {
        val prefs = context.getSharedPreferences(key, Context.MODE_PRIVATE)
        val gson = Gson()
        return prefs.getString(key, "")
    }

    fun SaveObject(context: Context, fileName: String, `object`: Any) {
        try {
            val fos = context.openFileOutput(fileName, Context.MODE_PRIVATE)
            val os: ObjectOutputStream
            os = ObjectOutputStream(fos)
            os.writeObject(`object`)
            os.close()
            fos.close()
        } catch (e: IOException) {
            e.message;
        }
    }

    fun LoadObject(context: Context, fileName: String): Any? {
        try {
            val fis = context.openFileInput(fileName)
            val `is` = ObjectInputStream(fis)
            val o = `is`.readObject()
            `is`.close()
            fis.close()

            return o
        } catch (e: Exception) {

        }

        return null
    }
}
